vortex-array/src/stats/precision.rs

```diff
@@ -0,0 +1,139 @@
+use std::fmt::{Debug, Display, Formatter};
+
+use vortex_dtype::DType;
+use vortex_error::VortexResult;
+use vortex_scalar::{Scalar, ScalarValue};
+
+use crate::stats::precision::Precision::{Exact, Inexact};
+
+/// A statistic has a precision `Exact` or `Inexact`. This represents uncertainty in that value.
+/// Exact values are computed, where can inexact values are likely inferred from compute functions.
+///
+/// Inexact statistics form a range of possible values that the statistic could be.
+/// This is statistic specific, for max this will be an upper bound. Meaning that the actual max
+/// in an array is guaranteed to be less than or equal to the inexact value, but equal to the exact
+/// value.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Precision<T> {
+    Exact(T),
+    Inexact(T),
+}
+
+impl<T> Precision<Option<T>> {
+    pub fn transpose(self) -> Option<Precision<T>> {
+        match self {
+            Exact(Some(x)) => Some(Exact(x)),
+            Inexact(Some(x)) => Some(Inexact(x)),
+            _ => None,
+        }
+    }
+}
+
+impl<T: Clone> Precision<T> {
+    pub fn into_inexact(self) -> Self {
+        match self {
+            Exact(val) => Inexact(val),
+            Inexact(_) => self,
+        }
+    }
+}
+
+impl<T> Precision<T> {
+    pub fn exact<S: Into<T>>(s: S) -> Precision<T> {
+        Exact(s.into())
+    }
+
+    pub fn inexact<S: Into<T>>(s: S) -> Precision<T> {
+        Inexact(s.into())
+    }
+
+    pub fn as_ref(&self) -> Precision<&T> {
+        match self {
+            Exact(val) => Exact(val),
+            Inexact(val) => Inexact(val),
+        }
+    }
+
+    pub fn ok_exact(self) -> Option<T> {
```

Comment:
`ok` should really only be used in the context of errors.

`some_exact` or even just `exact` (where it's then clear the result is an Option)